include/xtensor/xstrided_view.hpp

```diff
@@ -815,18 +860,19 @@ namespace xt
             "traversal has to be row or column major"
         );
 
-        using shape_type = std::decay_t<S>;
```

Comment:
@JohanMabille I think this will make the assertion at test_xstrided_view.cpp(709) fail. If I pass a signed shape into the view then the resulting expression will have a signed shape and the static shape types won't match. I'm not sure this is a meaningful assertion as long as they're both integral types.

Reply:
Oh, actually we have an inconsistency between [arbitrary shape type](https://github.com/xtensor-stack/xtensor/blob/5f49f6407501474cad622f9f16da172ab3717de3/include/xtensor/xstrided_view.hpp#L848) and [array shape type](https://github.com/xtensor-stack/xtensor/blob/5f49f6407501474cad622f9f16da172ab3717de3/include/xtensor/xstrided_view.hpp#L862): for the latter one, we convert the shape to an unsigned one when necessary, thus the failure in the test when you pass a signed one.

In order to support `-1` in the shape, we need to remove this conversion and forward the shape as `using shape_type = std::array<I, N>;` instead of `using shape_type = std::array<std::size_t, N>;`. Failing tests should been fixed accordingly.

Reply:
Notice that if we want to ensure the `shape_type` of the resulting `strided_view` to be unsigned (which we should probably do), you can add the following before instantiating the view:
```cpp
using unsigned_shape_type = rebind_container_t<shape_type>;
```

and then forward the shape with this new type:

```cpp
return view_type(std::forward<E>(e), xtl::forward_sequence<unsigned_shape_type, shape_type>(shape), std::move(strides), 0, e.layout());
```